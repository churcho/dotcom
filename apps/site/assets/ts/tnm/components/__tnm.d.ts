// eslint-disable-next-line typescript/no-unused-vars
import { Route, Stop } from "../../__v3api";

export type TNMMode = "subway" | "bus" | "rail";

export interface TNMRoute extends Route {
  stops: TNMStop[];
}

export interface TNMStop extends Stop {
  directions: TNMDirection[];
}

type DirectionId = 0 | 1;

export interface TNMDirection {
  direction_id: DirectionId;
  headsigns: TNMHeadsign[];
}

export interface TNMHeadsign {
  name: string;
  times: TNMTime[];
  train_number: string | null;
}

export interface TNMTime {
  delay: number;
  scheduled_time: string[] | null;
  prediction: TNMPrediction | null;
}

export interface TNMPrediction {
  time: string[];
  status: string | null;
  track: string | null;
}

export interface Trip {
  "bikes_allowed?": true;
  direction_id: 0 | 1;
  headsign: string;
  id: string;
  name: string;
  shape_id: string;
}

export interface Schedule {
  "early_departure?": boolean;
  "flag?": boolean;
  pickup_type: number;
  route: Route;
  stop: TNMStop;
  stop_sequence: number;
  time: string;
  trip: Trip;
}

interface StopWithRoutes {
  distance: string;
  stop: TNMStop;
  routes: RouteGroup[];
}

type RouteGroupName = "commuter_rail" | "subway" | "bus" | "ferry";

interface RouteGroup {
  group_name: RouteGroupName;
  routes: TNMRoute[];
}
