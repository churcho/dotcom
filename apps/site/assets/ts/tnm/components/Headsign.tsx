import React, { ReactElement } from "react";
import { RouteType, TNMHeadsign, TNMTime, TNMPrediction } from "./__tnm";

interface Props {
  headsign: TNMHeadsign;
  routeType: RouteType;
  condensed: boolean;
}

const Headsign = (props: Props): ReactElement<any> => {
  const { headsign, routeType, condensed } = props;
  return (
    <div className={headsignClass(condensed)}>
      <div className="m-tnm-sidebar__headsign">
        {renderHeadsignName(props)}

        {routeType === 2 && renderTrainName(`Train ${headsign.train_number}`)}
      </div>
      <div className="m-tnm-sidebar__schedules">
        {headsign.times.map(time => renderTime(time, headsign.name, routeType))}
      </div>
    </div>
  );
};

const headsignClass = (condensed: boolean): string => {
  if (condensed === true) {
    return "m-tnm-sidebar__headsign-schedule m-tnm-sidebar__headsign-schedule--condensed";
  }
  return "m-tnm-sidebar__headsign-schedule";
};

const renderHeadsignName = ({
  headsign,
  routeType,
  condensed
}: Props): ReactElement<any> => {
  const modifier = !condensed && routeType == 3 ? "small" : "large";

  const headsignNameClass = `m-tnm-sidebar__headsign-name m-tnm-sidebar__headsign-name--${modifier}`;

  if (headsign.name.includes(" via ")) {
    const split = headsign.name.split(" via ");
    return (
      <>
        <div className={headsignNameClass}>{split[0]}</div>
        <div className="m-tnm-sidebar__via">{`via ${split[1]}`}</div>
      </>
    );
  }
  return <div className={headsignNameClass}>{headsign.name}</div>;
};

const renderTrainName = (trainName: string): ReactElement<any> => (
  <div className="m-tnm-sidebar__headsign-train">{trainName}</div>
);

const renderTime = (
  tnmTime: TNMTime,
  headsignName: string,
  routeType: RouteType
) => {
  const { prediction, scheduled_time } = tnmTime;
  const time = prediction ? prediction.time : scheduled_time;

  return (
    <div
      key={`${headsignName}-${scheduled_time.join("")}`}
      className="m-tnm-sidebar__schedule"
    >
      {routeType === 2
        ? renderTimeCommuterRail(time, prediction)
        : renderTimeDefault(time)}
    </div>
  );
};

const renderTimeDefault = (time: Array<string>): ReactElement<any> => {
  const [num, _space, mins] = time;
  return (
    <div className="m-tnm-sidebar__time">
      <div className="m-tnm-sidebar__time-number">{num}</div>
      <div className="m-tnm-sidebar__time-mins">{mins}</div>
    </div>
  );
};

const renderTimeCommuterRail = (
  time: Array<string>,
  prediction: TNMPrediction | null
): ReactElement<any> => (
  <div className="m-tnm-sidebar__time m-tnm-sidebar__time--commuter-rail">
    <div className="m-tnm-sidebar__time-number">{time.join("")}</div>
    <div className="m-tnm-sidebar__status">
      {prediction ? prediction.status : "On Time"}
    </div>
  </div>
);

export default Headsign;
