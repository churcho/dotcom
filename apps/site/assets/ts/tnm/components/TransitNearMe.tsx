import React, { useReducer, ReactElement } from "react";
import TransitNearMeMap from "./googleMaps/TransitNearMeMap";
import RoutesSidebar from "./RoutesSidebar";
import { Route, Stop, MapData } from "./__tnm";
import { reducer, initialState, SelectedStopType } from "../state";

interface Props {
  mapData: MapData;
  mapId: string;
  sidebarData: Route[];
  getSidebarOffset: () => number;
}

const selectedStop = (
  sidebarData: Route[],
  selectedStopId: SelectedStopType
): Stop | undefined => {
  const stops = sidebarData.reduce(
    (acc, route) => acc.concat(route.stops),
    [] as Stop[]
  );
  return stops.find(stop => stop.id === selectedStopId);
};

const TransitNearMe = ({
  mapData,
  mapId,
  sidebarData,
  getSidebarOffset
}: Props): ReactElement<HTMLElement> => {
  const [state, dispatch] = useReducer(reducer, initialState);

  return (
    <div className="m-tnm">
      <div
        id={mapId}
        className="m-tnm__map"
        role="application"
        aria-label="Map with stops"
      />
      <TransitNearMeMap
        selectedStopId={state.selectedStopId}
        dispatch={dispatch}
        mapElementId={mapId}
        initialData={mapData}
        shouldCenterMapOnSelectedStop={state.shouldCenterMapOnSelectedStop}
      />
      <RoutesSidebar
        selectedStop={selectedStop(sidebarData, state.selectedStopId)}
        selectedStopId={state.selectedStopId}
        dispatch={dispatch}
        data={sidebarData}
        shouldFilterStopCards={state.shouldFilterStopCards}
        getOffset={getSidebarOffset}
      />
    </div>
  );
};

export default TransitNearMe;
