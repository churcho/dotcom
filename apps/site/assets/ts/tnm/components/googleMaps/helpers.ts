// @ts-ignore
import stopMarker from "../../../../static/images/icon-map-stop-marker.svg";
// @ts-ignore
import stationMarker from "../../../../static/images/icon-map-station-marker.svg";
// @ts-ignore
import selectedStopMarker from "../../../../static/images/icon-map-stop-marker-hover.svg";
// @ts-ignore
import selectedStationMarker from "../../../../static/images/icon-map-station-marker-hover.svg";
// @ts-ignore
import currentLocationMarker from "../../../../static/images/icon-current-location-marker.svg";
import { iconSize } from "../../../../js/google-map/helpers";

import { MarkerData } from "../__tnm";

interface GoogleMapMarkerSVG {
  url: string;
  scaledSize: google.maps.Size;
  origin: google.maps.Point;
  anchor: google.maps.Point;
  labelOrigin: google.maps.Point;
}

export const stopIcon = (selected: boolean): string =>
  selected ? selectedStopMarker : stopMarker;
export const stationIcon = (selected: boolean): string =>
  selected ? selectedStationMarker : stationMarker;

export const buildIconFromSVG = (
  svg: string,
  size: string
): GoogleMapMarkerSVG => {
  const sizeInt = iconSize(size);

  const encoded = window.btoa(svg);

  return {
    url: `data:image/svg+xml;base64, ${encoded}`,
    scaledSize: new window.google.maps.Size(sizeInt, sizeInt),
    origin: new window.google.maps.Point(0, 0),
    anchor: new window.google.maps.Point(sizeInt / 2, sizeInt / 2),
    labelOrigin: new window.google.maps.Point(sizeInt / 2, sizeInt / 2)
  };
};

export const buildMarkerIcon = (
  data: MarkerData,
  selected: boolean
): GoogleMapMarkerSVG | undefined => {
  if (data.icon) {
    if (data.icon!.includes("station")) {
      return buildIconFromSVG(stationIcon(selected), data.size);
    }
    if (data.icon!.includes("stop")) {
      return buildIconFromSVG(stopIcon(selected), data.size);
    }
    if (data.icon!.includes("current-location")) {
      return buildIconFromSVG(currentLocationMarker, data.size);
    }
  }
  return undefined;
};
