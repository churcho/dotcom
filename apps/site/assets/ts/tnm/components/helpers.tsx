import React, { KeyboardEvent } from "react";

export const handleEnterKeyPress = (
  e: KeyboardEvent,
  onClick: Function
): void => {
  if (e.key === "Enter") {
    onClick();
  }
};

export const renderSvg = (className: string, svgText: string): JSX.Element => (
  <span
    className={className}
    // eslint-disable-next-line react/no-danger
    dangerouslySetInnerHTML={{ __html: svgText }}
  />
);
