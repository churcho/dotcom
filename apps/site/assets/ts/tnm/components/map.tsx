import React from "react";
import ReactDOM from "react-dom";
// @ts-ignore: Not typed
import GoogleMap from "../../../js/google-map-class";

interface Props {
  initialData: any;
  mapElementId: string;
}

class TransitNearMeMap extends React.Component<Props> {
  public componentDidMount(): void {
    const { mapElementId, initialData } = this.props;
    this.map = new GoogleMap(mapElementId, initialData);
    this.map.bind(this);
    if (initialData.markers.length > 0) {
      this.tightenBounds();
    }
  }

  /* eslint-disable typescript/no-explicit-any */
  public data: any;

  public map: any;
  /* eslint-enable typescript/no-explicit-any */

  public tightenBounds(): void {
    this.map.resetBounds(["current-location", "radius-west", "radius-east"]);
  }

  public render(): null {
    return null;
  }
}

const render = (): void => {
  const dataEl = document.getElementById("js-tnm-map-dynamic-data");
  if (dataEl) {
    const id = dataEl.getAttribute("data-for") as string;
    const initialData = JSON.parse(dataEl.innerHTML);
    ReactDOM.render(
      <div className="m-tnm__map">
        <div id={id} />
        <TransitNearMeMap mapElementId={id} initialData={initialData} />
      </div>,
      document.getElementById("react-root")
    );
  }
};

export default render;
