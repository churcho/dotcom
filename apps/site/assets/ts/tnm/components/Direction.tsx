import React, { ReactElement } from "react";
import Headsign from "./Headsign";
import { TNMDirection, TNMRoute } from "./__tnm";

interface Props {
  direction: TNMDirection;
  route: TNMRoute;
  stopId: string;
}

export const directionIsEmpty = (dir: TNMDirection): boolean =>
  dir.headsigns.length === 0;

export const Direction = ({
  direction,
  route
}: Props): ReactElement<HTMLElement> | null => {
  if (directionIsEmpty(direction)) {
    return null;
  }

  const condensed = direction.headsigns.length === 1;

  const hideDirectionDestination =
    condensed || route.type === 0 || route.type === 1 || route.type === 2;

  return (
    <div>
      <div className="m-tnm-sidebar__direction">
        <div className="m-tnm-sidebar__direction-name u-small-caps">
          {route.direction_names[direction.direction_id]}
        </div>
        {!hideDirectionDestination && (
          <div className="m-tnm-sidebar__direction-destination">
            {route.direction_destinations[direction.direction_id]}
          </div>
        )}
      </div>
      {direction.headsigns.map(headsign => (
        <Headsign
          key={headsign.name}
          headsign={headsign}
          routeType={route.type}
          condensed={condensed}
        />
      ))}
    </div>
  );
};

export default Direction;
