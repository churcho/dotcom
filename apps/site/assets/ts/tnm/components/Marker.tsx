import React, { ReactElement } from "react";
import MapMarker from "../../app/googleMaps/Marker";
import stopIncludesModes from "../helpers/stop-includes-modes";
import { MarkerData } from "../../app/googleMaps/__googleMaps";
import { Mode } from "../../__v3api";
import { Dispatch } from "../state";
import { StopWithRoutes } from "./__tnm";

interface Props {
  map: google.maps.Map;
  data: MarkerData;
  isSelected: boolean;
  dispatch: Dispatch;
  stopData: StopWithRoutes | undefined;
  shouldFilterMarkers: boolean;
  selectedModes: Mode[];
}

export const isVisible = (
  stopData: StopWithRoutes | undefined,
  shouldFilterMarkers: boolean,
  selectedModes: Mode[]
): boolean =>
  shouldFilterMarkers && stopData
    ? stopIncludesModes(stopData, selectedModes)
    : true;

const Marker = ({
  map,
  data,
  isSelected,
  dispatch,
  stopData,
  shouldFilterMarkers,
  selectedModes
}: Props): ReactElement<HTMLElement> => (
  <MapMarker
    map={map}
    data={{
      ...data,
      "visible?": isVisible(stopData, shouldFilterMarkers, selectedModes)
    }}
    dispatch={dispatch}
    isSelected={isSelected}
  />
);

export default Marker;
