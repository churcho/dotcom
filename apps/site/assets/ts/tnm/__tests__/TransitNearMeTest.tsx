import React from "react";
import renderer from "react-test-renderer";
import { mount } from "enzyme";
import TransitNearMe, {
  getSelectedStop,
  fetchData
} from "../components/TransitNearMe";
import { createReactRoot } from "../../app/helpers/testUtils";
import { importData, importStopData } from "./helpers/testUtils";
import { MapData } from "../../leaflet/components/__mapdata";

it("it renders", () => {
  /* eslint-disable typescript/camelcase */
  const mapData: MapData = {
    zoom: 14,
    width: 630,
    polylines: [],
    markers: [],
    default_center: { latitude: 0, longitude: 0 },
    height: 500,
    tile_server_url: ""
  };
  /* eslint-enable typescript/camelcase */

  const mapId = "test";
  const routeSidebarData = importData().slice(0, 3);
  const stopSidebarData = importStopData();

  createReactRoot();

  const tree = renderer
    .create(
      <TransitNearMe
        query={{}}
        mapData={mapData}
        mapId={mapId}
        routeSidebarData={routeSidebarData}
        stopSidebarData={stopSidebarData}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("it switches view on click", () => {
  /* eslint-disable typescript/camelcase */
  const mapData: MapData = {
    zoom: 14,
    width: 630,
    polylines: [],
    markers: [],
    default_center: { latitude: 0, longitude: 0 },
    height: 500,
    tile_server_url: ""
  };
  /* eslint-enable typescript/camelcase */

  const mapId = "test";
  const routeSidebarData = importData().slice(0, 3);
  const stopSidebarData = importStopData();

  createReactRoot();

  const wrapper = mount(
    <TransitNearMe
      query={{}}
      mapData={mapData}
      mapId={mapId}
      routeSidebarData={routeSidebarData}
      stopSidebarData={stopSidebarData}
    />
  );
  wrapper.find(".m-tnm-sidebar__view-change").simulate("click");
  expect(wrapper.exists("#tnm-sidebar-by-stops")).toBeTruthy();
});

it("sets mode filters based on query", () => {
  /* eslint-disable typescript/camelcase */
  const mapData: MapData = {
    zoom: 14,
    width: 630,
    polylines: [],
    markers: [],
    default_center: { latitude: 0, longitude: 0 },
    height: 500,
    tile_server_url: ""
  };
  /* eslint-enable typescript/camelcase */

  const mapId = "test";
  const routeSidebarData = importData();
  const stopSidebarData = importStopData();

  createReactRoot();

  const noFilter = mount(
    <TransitNearMe
      query={{}}
      mapData={mapData}
      mapId={mapId}
      routeSidebarData={routeSidebarData}
      stopSidebarData={stopSidebarData}
    />
  );
  const allModes = noFilter
    .find(".m-tnm-sidebar__route")
    .map(card => card.prop("data-mode"));
  expect(allModes).toHaveLength(22);
  expect(allModes).toEqual([
    "bus",
    "bus",
    "bus",
    "bus",
    "subway",
    "subway",
    "subway",
    "subway",
    "bus",
    "bus",
    "bus",
    "bus",
    "bus",
    "subway",
    "subway",
    "bus",
    "bus",
    "bus",
    "commuter_rail",
    "commuter_rail",
    "commuter_rail",
    "commuter_rail"
  ]);

  const withFilter = mount(
    <TransitNearMe
      query={{ filter: "commuter_rail" }}
      mapData={mapData}
      mapId={mapId}
      routeSidebarData={routeSidebarData}
      stopSidebarData={stopSidebarData}
    />
  );

  const rail = withFilter
    .find(".m-tnm-sidebar__route")
    .map(card => card.prop("data-mode"));

  expect(rail).toEqual([
    "commuter_rail",
    "commuter_rail",
    "commuter_rail",
    "commuter_rail"
  ]);
});

it("getSelectedStop returns the stop if found", () => {
  const data = importStopData();
  expect(getSelectedStop(data, data[0].stop.stop.id)).toEqual(
    data[0].stop.stop
  );
});

it("getSelectedStop returns undefined if not found", () => {
  const data = importStopData();
  expect(getSelectedStop(data, "unknown")).toEqual(undefined);
});

describe("fetchData", () => {
  it("fetches data", () => {
    window.fetch = jest.fn().mockImplementation(
      () =>
        new Promise((resolve: Function) =>
          resolve({
            json: () => [],
            ok: true,
            status: 200,
            statusText: "OK"
          })
        )
    );

    const spy = jest.fn();
    return fetchData({ latitude: "41.0", longitude: "-71.0" }, spy).then(() => {
      expect(window.fetch).toHaveBeenCalledWith(
        "/transit-near-me/api?latitude=41.0&longitude=-71.0"
      );
      expect(spy).toHaveBeenCalledWith({
        payload: { data: [] },
        type: "UPDATE_ROUTE_SIDEBAR_DATA"
      });
    });
  });

  it("fails gracefully when request fails", () => {
    window.fetch = jest.fn().mockImplementation(
      () =>
        new Promise((resolve: Function) =>
          resolve({
            json: () => "Internal Server Error",
            ok: false,
            status: 500,
            statusText: "INTERNAL SERVER ERROR"
          })
        )
    );

    const spy = jest.fn();
    return fetchData({ latitude: "41.0", longitude: "-71.0" }, spy).then(() => {
      expect(spy).not.toHaveBeenCalled();
    });
  });
  it("fetches data with Phoenix form param structure", () => {
    window.fetch = jest.fn().mockImplementation(
      () =>
        new Promise((resolve: Function) =>
          resolve({
            json: () => [],
            ok: true,
            status: 200,
            statusText: "OK"
          })
        )
    );
    const spy = jest.fn();
    return fetchData(
      { "location[latitude]": "41.0", "location[longitude]": "-71.0" },
      spy
    ).then(() => {
      expect(window.fetch).toHaveBeenCalledWith(
        "/transit-near-me/api?location[latitude]=41.0&location[longitude]=-71.0"
      );
      expect(spy).toHaveBeenCalledWith({
        payload: { data: [] },
        type: "UPDATE_ROUTE_SIDEBAR_DATA"
      });
    });
  });

  it("does not fetch data if query doesn't have lat/lng", () => {
    const spy = jest.fn();
    return fetchData({}, spy).then(() => {
      expect(spy).not.toHaveBeenCalled();
    });
  });
});
