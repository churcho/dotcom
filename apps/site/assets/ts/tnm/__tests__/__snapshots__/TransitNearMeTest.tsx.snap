// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders 1`] = `
<div
  className="m-tnm"
>
  <div
    aria-label="Map with stops"
    className="m-tnm__map"
    id="test"
    role="application"
  />
  <div
    className="m-tnm-sidebar"
    style={
      Object {
        "left": "3px",
      }
    }
  >
    <div
      className="m-tnm-sidebar__header"
    >
      <h2>
        Nearby Routes
      </h2>
    </div>
    <div
      className="m-tnm-sidebar__route"
    >
      <div
        className="h3 m-tnm-sidebar__route-name u-bg--bus"
      >
        <span
          className="bus-route-sign"
        >
          504
        </span>
      </div>
      <div
        className="m-tnm-sidebar__stop"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex={0}
      >
        <div
          className="m-tnm-sidebar__stop-info"
        >
          <a
            className="m-tnm-sidebar__stop-name"
            href="/stops/9983"
          >
            <span
              className="m-tnm-sidebar__stop-marker"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "",
                }
              }
            />
             
            Stuart St @ Charles St S
          </a>
          <div
            className="m-tnm-sidebar__stop-distance"
          >
            238 ft
          </div>
        </div>
        <div>
          <div
            className="m-tnm-sidebar__direction"
          >
            <div
              className="m-tnm-sidebar__direction-name u-small-caps"
            >
              Outbound
            </div>
          </div>
          <div
            className="m-tnm-sidebar__headsign-schedule m-tnm-sidebar__headsign-schedule--condensed"
          >
            <div
              className="m-tnm-sidebar__headsign"
            >
              <div
                className="m-tnm-sidebar__headsign-name m-tnm-sidebar__headsign-name--large"
              >
                Watertown
              </div>
              <div
                className="m-tnm-sidebar__via"
              >
                via Copley (Express)
              </div>
            </div>
            <div
              className="m-tnm-sidebar__schedules"
            >
              <div
                className="m-tnm-sidebar__schedule"
              >
                <div
                  className="m-tnm-sidebar__time"
                >
                  <div
                    className="m-tnm-sidebar__time-number"
                  >
                    14
                  </div>
                  <div
                    className="m-tnm-sidebar__time-mins"
                  >
                    min
                  </div>
                </div>
              </div>
              <div
                className="m-tnm-sidebar__schedule"
              >
                <div
                  className="m-tnm-sidebar__time"
                >
                  <div
                    className="m-tnm-sidebar__time-number"
                  >
                    44
                  </div>
                  <div
                    className="m-tnm-sidebar__time-mins"
                  >
                    min
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="m-tnm-sidebar__stop"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex={0}
      >
        <div
          className="m-tnm-sidebar__stop-info"
        >
          <a
            className="m-tnm-sidebar__stop-name"
            href="/stops/6542"
          >
            <span
              className="m-tnm-sidebar__stop-marker"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "",
                }
              }
            />
             
            Kneeland St @ Washington St
          </a>
          <div
            className="m-tnm-sidebar__stop-distance"
          >
            0.2 mi
          </div>
        </div>
        <div>
          <div
            className="m-tnm-sidebar__direction"
          >
            <div
              className="m-tnm-sidebar__direction-name u-small-caps"
            >
              Inbound
            </div>
          </div>
          <div
            className="m-tnm-sidebar__headsign-schedule m-tnm-sidebar__headsign-schedule--condensed"
          >
            <div
              className="m-tnm-sidebar__headsign"
            >
              <div
                className="m-tnm-sidebar__headsign-name m-tnm-sidebar__headsign-name--large"
              >
                Downtown
              </div>
              <div
                className="m-tnm-sidebar__via"
              >
                via Copley (Express)
              </div>
            </div>
            <div
              className="m-tnm-sidebar__schedules"
            >
              <div
                className="m-tnm-sidebar__schedule"
              >
                <div
                  className="m-tnm-sidebar__time"
                >
                  <div
                    className="m-tnm-sidebar__time-number"
                  >
                    1
                  </div>
                  <div
                    className="m-tnm-sidebar__time-mins"
                  >
                    min
                  </div>
                </div>
              </div>
              <div
                className="m-tnm-sidebar__schedule"
              >
                <div
                  className="m-tnm-sidebar__time"
                >
                  <div
                    className="m-tnm-sidebar__time-number"
                  >
                    14
                  </div>
                  <div
                    className="m-tnm-sidebar__time-mins"
                  >
                    min
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="m-tnm-sidebar__route"
    >
      <div
        className="h3 m-tnm-sidebar__route-name u-bg--bus"
      >
        <span
          className="bus-route-sign"
        >
          55
        </span>
      </div>
      <div
        className="m-tnm-sidebar__stop"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex={0}
      >
        <div
          className="m-tnm-sidebar__stop-info"
        >
          <a
            className="m-tnm-sidebar__stop-name"
            href="/stops/9983"
          >
            <span
              className="m-tnm-sidebar__stop-marker"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "",
                }
              }
            />
             
            Stuart St @ Charles St S
          </a>
          <div
            className="m-tnm-sidebar__stop-distance"
          >
            238 ft
          </div>
        </div>
        <div>
          <div
            className="m-tnm-sidebar__direction"
          >
            <div
              className="m-tnm-sidebar__direction-name u-small-caps"
            >
              Outbound
            </div>
          </div>
          <div
            className="m-tnm-sidebar__headsign-schedule m-tnm-sidebar__headsign-schedule--condensed"
          >
            <div
              className="m-tnm-sidebar__headsign"
            >
              <div
                className="m-tnm-sidebar__headsign-name m-tnm-sidebar__headsign-name--large"
              >
                Queensberry St
              </div>
            </div>
            <div
              className="m-tnm-sidebar__schedules"
            >
              <div
                className="m-tnm-sidebar__schedule"
              >
                <div
                  className="m-tnm-sidebar__time"
                >
                  <div
                    className="m-tnm-sidebar__time-number"
                  >
                    39
                  </div>
                  <div
                    className="m-tnm-sidebar__time-mins"
                  >
                    min
                  </div>
                </div>
              </div>
              <div
                className="m-tnm-sidebar__schedule"
              >
                <div
                  className="m-tnm-sidebar__time"
                >
                  <div
                    className="m-tnm-sidebar__time-number"
                  >
                    12:55
                  </div>
                  <div
                    className="m-tnm-sidebar__time-mins"
                  >
                    PM
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="m-tnm-sidebar__stop"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex={0}
      >
        <div
          className="m-tnm-sidebar__stop-info"
        >
          <a
            className="m-tnm-sidebar__stop-name"
            href="/stops/8279"
          >
            <span
              className="m-tnm-sidebar__stop-marker"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "",
                }
              }
            />
             
            Tremont St @ Boylston Station
          </a>
          <div
            className="m-tnm-sidebar__stop-distance"
          >
            0.2 mi
          </div>
        </div>
        <div>
          <div
            className="m-tnm-sidebar__direction"
          >
            <div
              className="m-tnm-sidebar__direction-name u-small-caps"
            >
              Outbound
            </div>
          </div>
          <div
            className="m-tnm-sidebar__headsign-schedule m-tnm-sidebar__headsign-schedule--condensed"
          >
            <div
              className="m-tnm-sidebar__headsign"
            >
              <div
                className="m-tnm-sidebar__headsign-name m-tnm-sidebar__headsign-name--large"
              >
                Queensberry St
              </div>
            </div>
            <div
              className="m-tnm-sidebar__schedules"
            >
              <div
                className="m-tnm-sidebar__schedule"
              >
                <div
                  className="m-tnm-sidebar__time"
                >
                  <div
                    className="m-tnm-sidebar__time-number"
                  >
                    39
                  </div>
                  <div
                    className="m-tnm-sidebar__time-mins"
                  >
                    min
                  </div>
                </div>
              </div>
              <div
                className="m-tnm-sidebar__schedule"
              >
                <div
                  className="m-tnm-sidebar__time"
                >
                  <div
                    className="m-tnm-sidebar__time-number"
                  >
                    12:55
                  </div>
                  <div
                    className="m-tnm-sidebar__time-mins"
                  >
                    PM
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="m-tnm-sidebar__route"
    >
      <div
        className="h3 m-tnm-sidebar__route-name u-bg--bus"
      >
        <span
          className="bus-route-sign"
        >
          553
        </span>
      </div>
      <div
        className="m-tnm-sidebar__stop"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex={0}
      >
        <div
          className="m-tnm-sidebar__stop-info"
        >
          <a
            className="m-tnm-sidebar__stop-name"
            href="/stops/9983"
          >
            <span
              className="m-tnm-sidebar__stop-marker"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "",
                }
              }
            />
             
            Stuart St @ Charles St S
          </a>
          <div
            className="m-tnm-sidebar__stop-distance"
          >
            238 ft
          </div>
        </div>
        <div>
          <div
            className="m-tnm-sidebar__direction"
          >
            <div
              className="m-tnm-sidebar__direction-name u-small-caps"
            >
              Outbound
            </div>
          </div>
          <div
            className="m-tnm-sidebar__headsign-schedule m-tnm-sidebar__headsign-schedule--condensed"
          >
            <div
              className="m-tnm-sidebar__headsign"
            >
              <div
                className="m-tnm-sidebar__headsign-name m-tnm-sidebar__headsign-name--large"
              >
                Waltham
              </div>
              <div
                className="m-tnm-sidebar__via"
              >
                via Copley (Express)
              </div>
            </div>
            <div
              className="m-tnm-sidebar__schedules"
            >
              <div
                className="m-tnm-sidebar__schedule"
              >
                <div
                  className="m-tnm-sidebar__time"
                >
                  <div
                    className="m-tnm-sidebar__time-number"
                  >
                    7:07
                  </div>
                  <div
                    className="m-tnm-sidebar__time-mins"
                  >
                    PM
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="m-tnm-sidebar__stop"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex={0}
      >
        <div
          className="m-tnm-sidebar__stop-info"
        >
          <a
            className="m-tnm-sidebar__stop-name"
            href="/stops/6542"
          >
            <span
              className="m-tnm-sidebar__stop-marker"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "",
                }
              }
            />
             
            Kneeland St @ Washington St
          </a>
          <div
            className="m-tnm-sidebar__stop-distance"
          >
            0.2 mi
          </div>
        </div>
        <div>
          <div
            className="m-tnm-sidebar__direction"
          >
            <div
              className="m-tnm-sidebar__direction-name u-small-caps"
            >
              Inbound
            </div>
          </div>
          <div
            className="m-tnm-sidebar__headsign-schedule m-tnm-sidebar__headsign-schedule--condensed"
          >
            <div
              className="m-tnm-sidebar__headsign"
            >
              <div
                className="m-tnm-sidebar__headsign-name m-tnm-sidebar__headsign-name--large"
              >
                Downtown
              </div>
              <div
                className="m-tnm-sidebar__via"
              >
                via Copley (Express)
              </div>
            </div>
            <div
              className="m-tnm-sidebar__schedules"
            >
              <div
                className="m-tnm-sidebar__schedule"
              >
                <div
                  className="m-tnm-sidebar__time"
                >
                  <div
                    className="m-tnm-sidebar__time-number"
                  >
                    8:24
                  </div>
                  <div
                    className="m-tnm-sidebar__time-mins"
                  >
                    PM
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
